### PR TITLE
Config

### PR DESCRIPTION
**Pull request by miguelgomezg**

### CODE REVIEW

File: src/api.py [hunk 1]
```diff
@@ -3,10 +3,12 @@
...
+    units_routes, 
+    distributions_routes)
 
 app = FastAPI(**fastApiConfig)
 db = MongoEngine().get_connection()
 
-app.include_router(units_routes)
\ No newline at end of file
+app.include_router(units_routes)
+app.include_router(distributions_routes)
```
Comment (davisfelipe) [inline src/api.py:14]: Blank Line at the end of the line

File: src/interfaces/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from .distribution_interface import DistributionInterface
+
+__all__ = ["DistributionInterface"]
```
Comment (davisfelipe) [inline src/interfaces/__init__.py:3]: Blank Line at the end of the line

File: src/models/db/Distribution.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from mongoengine import (
+    Document,
+    StringField,
+    ListField
+)
+
+class Distribution(Document):
+    name = StringField(required=True)
+    type = ListField(required=True)
```
Comment (davisfelipe) [inline src/models/db/Distribution.py:9]: Blank Line at the end of the line

File: src/routes/__init__.py [hunk 4]
```diff
@@ -1,5 +1,7 @@
 from .units import units_routes
+from .distributions import distributions_routes
 
 __all__ = [
-    "units_routes"
+    "units_routes",
+    "distributions_routes"
 ]
```
Comment (davisfelipe) [inline src/routes/__init__.py:7]: Blank Line at the end of the line

File: src/routes/distributions.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    HTTP_200_OK,
+    HTTP_400_BAD_REQUEST,
+    HTTP_404_NOT_FOUND
+)
+
+from src.utils import (
+        BsonObject,
+        UJSONResponse
+        ,DistributionMessage)
+
```
Comment (davisfelipe) [inline src/routes/distributions.py:12]: Sort Imports
```suggestion
from src.utils import (
        BsonObject,
        UJSONResponse,
        DistributionMessage
 )

```

File: src/routes/distributions.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    '''
+        Get  all distributions
+    '''
+    distributions = DistributionInterface.find_all()
+
+    if not distributions:
+        return UJSONResponse(
+        DistributionMessage.not_found, 
+        HTTP_404_NOT_FOUND
+        )
```
Comment (davisfelipe) [inline src/routes/distributions.py:28]: Identation
```suggestion
        return UJSONResponse(
            DistributionMessage.not_found, 
            HTTP_404_NOT_FOUND
        )
```

File: src/routes/distributions.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        return UJSONResponse(
+        DistributionMessage.not_exist, 
+        HTTP_400_BAD_REQUEST
+        )  
+
+    return UJSONResponse(
+        DistributionMessage.found, 
+        HTTP_200_OK,
+        BsonObject.dict(parameters)
+    )
```
Comment (davisfelipe) [inline src/routes/distributions.py:56]: Blank Line at the end of the line

File: src/routes/test/test_units.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+from unittest import TestCase
+from fastapi.testclient import TestClient
```
Comment (davisfelipe) [inline src/routes/test/test_units.py:2]: Sort Import, remeber [PEP8](https://www.python.org/dev/peps/pep-0008/#imports)
```suggestion
from unittest import TestCase

from fastapi.testclient import TestClient
```

File: src/utils/__init__.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+from .messages import UnitsMessage, DistributionMessage
+from .response import UJSONResponse
+
+__all__ = [
+    "DateTime",
+    "BsonObject",
+    "UnitsMessage",
+    "DistributionMessage",
+    "UJSONResponse"
+]
```
Comment (davisfelipe) [inline src/utils/__init__.py:12]: Blank Line at the end of the line

File: src/utils/messages.py [hunk 10]
```diff
@@ -5,4 +5,13 @@ class UnitsMessage:
...
+    distance: str = "Units distance"
+
+@dataclass
+class DistributionMessage:
+    '''
+        Messages used in endpoint responses for distributions
+    '''
+    found: str = 'Found distributions'
+    not_found: str = 'Distributions not found'
+    not_exist: str = 'Distribution does not exist'
```
Comment (davisfelipe) [inline src/utils/messages.py:17]: Blank Line at the end of the line

File: src/api.py [hunk 11]
```diff
@@ -3,10 +3,12 @@
 
 from src.config import fastApiConfig
 from src.db import MongoEngine
-from src.routes import units_routes
-
+from src.routes import (
+    units_routes, 
+    distributions_routes)
```
Comment (davisfelipe) [inline src/api.py:8]:
```suggestion
from src.routes import (
    units_routes, 
    distributions_routes
)
```

File: src/interfaces/distribution_interface.py [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+            name (str): distribution name
+        Returns:
+            Distribution name and parameters
+        '''
+        filters = dict(
+            name=name
+        )
+        return Distribution.objects(
+                **filters
+                ).first()
```
Comment (davisfelipe) [inline src/interfaces/distribution_interface.py:31]:
```suggestion
        return Distribution.objects(
            **filters
        ).first()
```

File: src/routes/test/test_units.py [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+def solve_path(path: str):
+    source = 'src.routes.units'
+    return ".".join([source, path])
+
+class GetUnitsRouteTestCase(TestCase):
+    def setUp(self) -> None:
```
Comment (davisfelipe) [inline src/routes/test/test_units.py:9]: 2 blank spaces after base declarations
```suggestion
    return ".".join([source, path])


class GetUnitsRouteTestCase(TestCase):
    def setUp(self) -> None:
```

File: src/models/db/Simulation.py [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    iteration_time_units = StringField()
+    iteration_number = IntField()
+    box_size = ListField()
+    distance_units = StringField()
+    is_delete = BooleanField(default=False)
+
+class Simulation(BaseDocument):
+    identifer = UUIDField(unique=True)
+    configuration = ReferenceField(Configuration, dbref=True)
+    executed_at = DateField(required=True)
```
Comment (davisfelipe) [inline src/models/db/Simulation.py:30]: Blank Line at the end of the line

File: src/models/db/Simulation.py [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    ReferenceField,
+    DateField
+)
+from mongoengine.fields import BooleanField
+
+
+from .base import BaseDocument
+
+class Configuration(BaseDocument):
+    identifer = UUIDField(unique=True)
```
Comment (davisfelipe) [inline src/models/db/Simulation.py:17]: it could be named _uuid_, what do you think?

File: src/routes/test/test_units.py [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+        from src.api import app
+        self.client = TestClient(app)
+
+        self.route = "/units/distance"
+
+    def test_get_units_distance(self):
+        response = self.client.get(self.route)
+        self.assertIsNotNone(response)
+        self.assertEqual(response.status_code, 200)
```
Comment (davisfelipe) [inline src/routes/test/test_units.py:18]: Blank Line at the end of the line

**Review by davisfelipe — CHANGES_REQUESTED** [review]
Hi Miguel

I made suggestions in your code, tell if you need something

greetings

**Review by Camilo-HG — APPROVED** [review]

**Pull request merged**